addons/binding/org.openhab.binding.homepilot/src/main/java/org/openhab/binding/homepilot/internal/DefaultHttpGateway.java

```diff
@@ -0,0 +1,162 @@
+package org.openhab.binding.homepilot.internal;
+
+import static org.openhab.binding.homepilot.HomePilotBindingConstants.*;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.util.FormContentProvider;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.util.Fields;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Lists;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+public class DefaultHttpGateway implements HomePilotGateway {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultHttpGateway.class);
+
+    private String id;
+    private HomePilotConfig config;
+    private HttpClient httpClient;
+    // private Cache<String, String> responseCache;
+
+    public DefaultHttpGateway(String id, HomePilotConfig config) {
+        this.id = id;
+        this.config = config;
+    }
+
+    @Override
+    public String getId() {
+        return id;
+    }
+
+    @Override
+    public void initialize() {
+        httpClient = new HttpClient();
+        // httpClient.setConnectTimeout(config.getTimeout() * 1000L);
+
+        try {
+            httpClient.start();
+        } catch (Exception ex) {
+            throw new RuntimeException(ex.getMessage(), ex);
+        }
+
+        // responseCache = CacheBuilder.newBuilder().expireAfterWrite(10, TimeUnit.SECONDS)
+        // .build(new CacheLoader<String, String>() {
+        //
+        // @Override
+        // public String load(String url) throws Exception {
+        // logger.info("Calling url " + url);
+        // return httpClient.POST(url).header(HttpHeader.CONTENT_TYPE, "application/json;charset=utf-8")
+        // .send().getContentAsString();
+        // }
+        // });
+    }
+
+    @Override
+    public void cancelLoadAllDevices() {
+        // nothing to do here
+    }
+
+    @Override
+    public List<HomePilotDevice> loadAllDevices() {
+        String url = String.format("http://%s/deviceajax.do?alldevices", config.getAddress());
+        try {
+            return transform2Devices(
+                    httpClient.POST(url).header(HttpHeader.CONTENT_TYPE, "application/json;charset=utf-8").send()
+                            .getContentAsString()/* responseCache.get(url) */);
+        } catch (Exception e) {
```

Comment:
Does it throw Exception if not, please catch more specific and also throw more specific